hugr-py/src/hugr/tys.py

```diff
@@ -446,6 +483,10 @@ def resolve(self, registry: ext.ExtensionRegistry) -> PolyFuncType:
             body=self.body.resolve(registry),
         )
 
+    def __str__(self) -> str:
+        # ∀[a]. [list<a>] -> [Bool]
+        return f"∀{self.params}. {self.body!s})"
```

Comment:
This would look more like `∀[TypeTypeParam(TypeBound.Any)].[list<Variable(0)>] -> [Bool]`

We probably want more concise `__str__` implementations for `TypeParam`s. Also maybe we should print sth like `$0` instead of `Variable(0)` (same for `VariableArg`)?